### PR TITLE
[ci] install nvm locally; retool base dir

### PR DESCRIPTION
**Pull request by schallert**
- We can't make assumptions about the version of `node` a user may have
  installed. If we don't find `nvm`, install it locally.
- `retool` needs its base dir set if the nearest git dir isn't the
  project parent.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -419,19 +429,15 @@ node-yarn-run:
...
 else
-	node --version 2>&1 >/dev/null || \
-		(echo "Trying apt install" && which apt-get && (curl -sL https://deb.nodesource.com/setup_$(node_version).x | bash) && apt-get install -y nodejs) || \
-		(echo "Trying apk install" && which apk && apk add --update nodejs nodejs-npm) || \
-		(echo "No node install or known package manager" && exit 1)
-	@echo "Not using nvm, using node version $(shell node --version)"
-	bash -c "$(node_cmd)"
+	mkdir .nvm
+	# Install nvm locally
+	curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | NVM_DIR=$$(pwd)/.nvm PROFILE=/dev/null bash
```
Comment (prateek) [inline Makefile]: *dons tinfoil hat* not a fan of cURL+exec-ing , mind pulling the script into the repo

File: Makefile [hunk 2]
```diff
@@ -419,19 +429,15 @@ node-yarn-run:
...
+	bash -c "source $$(pwd)/.nvm/nvm.sh; nvm install 6"
```
Comment (prateek) [inline Makefile]: hm would `SELF_DIR` (declared at the top of the makefile) be better than `pwd`?

File: Makefile (hunk 1 above)
Comment (schallert) [reply Makefile]: Definitely :)

File: Makefile (hunk 2 above)
Comment (schallert) [reply Makefile]: Ah nice catch, ty

**Pull request merged**